packages/synapse-react-client/src/components/QueryContext/QueryContext.tsx

```diff
@@ -49,14 +48,17 @@ export type QueryContextType<
           : never
       >
     | undefined
+  currentQueryRequest: ReadonlyDeep<QueryBundleRequest>
+  nextQueryRequest: ReadonlyDeep<QueryBundleRequest>
```

Comment:
If consuming components just need to reference the query object, they can access this readonly object instead of calling `getCurrentQueryRequest`.

I think it's a bit more intuitive to use objects when thinking about the React component lifecycle, since it's more natural to consider cases where an object updates and triggers a re-render, vs a `get` function updating.